github/Artifact.py

```diff
@@ -0,0 +1,173 @@
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"id": self._id.value, "url": self._url.value})
+
+    @property
+    def id(self):
+        """
+        :type: integer
+        """
+        self._completeIfNotSet(self._id)
+        return self._id.value
+
+    @property
+    def node_id(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._node_id)
+        return self._node_id.value
+
+    @property
+    def name(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._name)
+        return self._name.value
+
+    @property
+    def size_in_bytes(self):
+        """
+        :type: integer
+        """
+        self._completeIfNotSet(self._size_in_bytes)
+        return self._size_in_bytes.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._url)
+        return self._url.value
+
+    @property
+    def archive_download_url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._archive_download_url)
+        return self._archive_download_url.value
+
+    @property
+    def expired(self):
+        """
+        :type: bool
+        """
+        self._completeIfNotSet(self._expired)
+        return self._expired.value
+
+    @property
+    def created_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._created_at)
+        return self._created_at.value
+
+    @property
+    def updated_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._updated_at)
+        return self._updated_at.value
+
+    @property
+    def expires_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._expires_at)
+        return self._expires_at.value
+
+    def download(self, archive_format="zip"):
+        """
+        :calls: `/repos/:owner/:repo/actions/artifacts/:artifact_id/:archive_format <https://developer.github.com/v3/actions/artifacts/#download-an-artifact>`_
+        :rtype: namedtuple with filename and content members
+        """
+        headers, data = self._requester.requestMultipartBinaryAndCheck(
+            "GET", self.url + "/" + archive_format
+        )
+        if headers.get("status") == "302 Found" and "location" in headers:
+            headers, data = self._requester.requestMultipartBinaryAndCheck(
+                "GET", headers["location"]
+            )
+
+        return data
+
+    def delete(self):
+        """
+        :calls: `DELETE /repos/:owner/:repo/actions/artifacts/:artifact_id <https://developer.github.com/v3/actions/artifacts/#delete-an-artifact>`_
+        :rtype: None
+        """
+        headers, data = self._requester.requestJsonAndCheck("DELETE", self.url)
```

Comment:
requestJson() and also check the returned status code is 204, like other examples in the code base.

Reply:
OK, sure. FYI, most of the things I found when I looked for `DELETE` were of this form, though? E.g.:

https://github.com/PyGithub/PyGithub/blob/1bb18ab52e9a5997774c9aed4e1521fe382e5b21/github/PullRequestComment.py#L172-L177

So that's probably why I ended up with this.

Reply:
I have changed it, at any rate; see 8ec83c32fa0be91d539a33803b0bddef99528f6d